python/tvm/relay/op/transform.py

```diff
@@ -1320,3 +1320,84 @@ def adv_index(inputs):
         Output tensor.
     """
     return _make.adv_index(Tuple(inputs))
+
+
+def sparsefillemptyrows(sparse_indices, sparse_values, dense_shape, default_value):
+    """
+    Fill first column of the empty rows with default values for a sparse array.
+
+    Parameters
+    ----------
+    sparse_indices : relay.Expr
+        A 2-D tensor[N, n_dim] of integers containing location of sparse values, where N is the
+        number of sparse values and n_dim is the number of dimensions of the dense_shape
+
+    sparse_values : relay.Expr
+        A 1-D tensor[N] containing the sparse values for the sparse indices.
+
+    dense_shape : relay.Expr
+        A list of integers. Shape of the dense output tensor.
+
+    default_value : relay.Expr
+        A 0-D tensor containing the default value for the remaining locations.
+        Defaults to 0.
+
+    Returns
+    -------
+    TupleWrapper with the following four outputs
+
+    new_sparse_indices : relay.Expr
+        A 2-D tensor[N + dense_shape[0], n_dim] of integers containing location of new sparse
+        indices where N is the number of sparse values. It is filled with -1 at to_be_discarded
+        indices.
+
+    empty_row_indicator : relay.Expr
+        A 1-D Boolean tensor[dense_shape[0]] indicating whether the particular row is empty
+
+    new_sparse_values : relay.Expr
+        A 1-D tensor[dense_shape[0]] containing the sparse values for the sparse indices. It is
+        filled with -1 at to_be_discarded indices.
+
+    slice_element_index : relay.Expr
+        A 1-D tensor containing the amount of elements in the sparse_indices and new_sparse_values
+        expression to be sliced in a future op discarding non-useful elements in new_sparse_indices
+        and new_sparse_values
+
+    Examples
+    -------
+
+    .. code-block:: python
+
+    sparse_indices = [[0, 1],
+                      [0, 3],
+                      [2, 0],
+                      [3, 1]]
+    sparse_values = [1, 2, 3, 4]
+    default_value = [10]
+    dense_shape = [5, 6]
+    new_sparse_indices, empty_row_indicator, new_sparse_values, slice_element_index =
+                        relay.sparsereshape(
+                        sparse_indices,
+                        sparse_values,
+                        prev_shape,
+                        new_shape)
+    new_sparse_indices =       [[0, 1],
+                               [0, 3],
+                               [2, 0],
+                               [3, 1],
+                               [1, 0],
+                               [4, 0],
+                               [-1, -1],
```

Comment:
I'm not sure this example is correct. `[-1, -1]` is not a valid sparse index. Shouldn't the output be
```
new_sparse_indices=[[0, 1],
                                      [0, 3]
                                      [1, 0]
                                      [2, 0]
                                      [3, 1]
                                      [4, 0]]

new_sparse_values = [1, 2, 10, 3, 4, 10]
```

Reply:
So , if you look at the description (`N + dense_shape[0]` ) <-- this is the dim0 shape of the tensor. This is because we can't know beforehand how many rows are missing. This operation would be succeeded by strided_slice to remove with begin as 0 and end as `slice_element_index` to achieve the desired output

Reply:
Let me know if you have any better suggestions for implementation, else I will stick with this.

Reply:
Ok, I understand now. Could you add a little more documentation to that effect? I would also fill the values associated invalid indices with zeros instead of -1. That way even if someone uses the indices, the effect would be nothing.

Also, have you considered using dynamic shape instead? That would avoid the need for the negative indices.

Reply:
No I haven't considered that. I noticed that for input dependent ops like `non_maximum_suppression`, they used nms with fixed size + strided slice and I just borrowed that idea. Do you mind pointing me to an implementation similar to what you are talking about ?

Reply:
We have it set up that way because that's how MXNet does it (TOPI's original opset was mostly cloned from MXNet), and MXNet is fairly conservative about static shapes. From a performance perspective, if the expectation is that we'll always use this with a dynamic strided slice after, I'd think it would be better to just emit a dynamic shape. If we have another usecase, however, I think this is fine.

Do you have a framework level usecase in mind?

Reply:
Yeah, in the short-term it will be used for TF. https://www.tensorflow.org/api_docs/python/tf/sparse/fill_empty_rows

Reply:
Given you are using it for a single op, I would follow NMS's example and use dynamic shape.

Reply:
Resolving this conversation since based on our offline discussion, we will start off with static shape